free.cabal

```diff
@@ -73,20 +73,25 @@ library
 
   build-depends:
     base                 == 4.*,
-    bifunctors           >= 4 && < 6,
     comonad              >= 4 && < 6,
     distributive         >= 0.2.1,
     mtl                  >= 2.0.1.0 && < 2.3,
     profunctors          >= 4 && < 6,
     semigroupoids        >= 4 && < 6,
-    semigroups           >= 0.8.3.1 && < 1,
     transformers         >= 0.2.0   && < 0.6,
     transformers-base    < 0.5,
-    transformers-compat  >= 0.3     && < 1,
     template-haskell     >= 2.7.0.0 && < 3,
     exceptions           >= 0.6 && < 0.11,
     containers           < 0.6
 
+  if impl(ghc < 8.2)
```

Comment:
Please, other way around. `!impl(ghc >= 8.0)`

See https://github.com/ekmett/recursion-schemes/pull/46

Reply:
Yes. Even the possibilities of something than GHC are small